tests/neg-macros/annot-codegen/Macro_1.scala

```diff
@@ -0,0 +1,83 @@
+import scala.annotation.{experimental, MacroAnnotation}
+import scala.quoted.*
+import scala.collection.mutable.ArrayBuffer
+
+@experimental
+class data extends MacroAnnotation:
+  import data.*
+  def transform(using Quotes)(tree: quotes.reflect.Definition): List[quotes.reflect.Definition] =
+    import quotes.reflect.*
+    tree match
+      case cdef: ClassDef =>
+        val classPatches = ArrayBuffer[String]()
+
+        val cls = tree.symbol
+        val expectedBody = s"data.generated[${cls.name}]()" // FIXME: handle type parameters
+
+        val params = paramNames(cls)
+        for param <- params do
+          val withParam = With(param)
+          val paramType = cls.declaredField(param).info
+          val existingOpt =
+            cls.declaredMethod(withParam).find(o =>
+              val paramss = o.paramSymss
+              paramss.size == 1 && paramss(0).size == 1
+              && paramss(0)(0).name == param // FIXME: if the parameter name is incorrect, propose rewriting it
+              && paramss(0)(0).info == paramType // FIXME: if the parameter type changed, propose rewriting it
+            )
+          existingOpt match
+            case Some(existing) => existing.tree match
+              case tree: DefDef =>
+                tree.rhs match
+                case Some(rhs) => rhs.asExpr match
+                  case '{data.generated[t]()} =>
+                    // The correct method is already present, nothing to do
+                  case _ =>
+                    report.error(s"Replace the underline code by:\n$expectedBody", rhs.pos)
+                case _ =>
+                  report.error("FIXME: Passing -Yretain-trees is currently needed for this macro to work")
```

Comment:
@nicolasstucki I'm surprised `-Yretain-trees` is needed since I'm accessing the tree of a symbol defined in the class being annotated, so the tree is accessible via traversing the `tree` argument of the macro annotation `transform` method, but it would be more convenient if I could access it through Symbol#tree too.